lib/API/JSON/v1/UserFlags.php

```diff
@@ -33,6 +33,7 @@ function set(array $flags = []) {
         ));
     } else {
       foreach ($flags as $name => $value) {
+        $value = htmlspecialchars($value);
```

Comment:
I am quite confused by this change. Could you please explain? In the commit message, you say this is filtering values, but that is not what you do here. You don't filter anything. 

I don't understand why do you call `htmlspecialchars` before saving flags to the database...

Reply:
Since all users are able to set flags, I thought it make sense to add protection against XSS.

Reply:
But in that case, you're doing it wrong. Protection against XSS is supposed to be done when displaying values not when saving them. When saving values we need to protect against SQL Injection. 

The reason is that you don't know in which context will the value used it. Is it going to be printed in JS code? in HTML code? In CSS code? We don't know that when saving, we know it when displaying. That is why protection against XSS is part of the templating system. 

But also we should keep values in the database unobfuscated so that we could use them in any context in the future and not tight them to a certain usage by calling a function that is tight to HTML.